src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -621,40 +625,24 @@ class CosmasGoogleCloudService(
                                     responseObserver: StreamObserver<CosmasProto.RestoreDeletedFileResponse>) = logging(
             "restoreDeletedFile", request.info.projectId, request.oldFileId,
             other = mapOf("newFileId" to request.newFileId)) {
-        val cemeteryName = "${request.info.projectId}-cemetery"
-        val cemeteryBytes: Blob? = try {
-            this.storage.get(getBlobId(cemeteryName, request.info))
-        } catch (e: StorageException) {
-            handleStorageException(e, responseObserver)
-            return@logging
-        }
-        val cemetery = if (cemeteryBytes == null) {
-            CosmasProto.FileCemetery.newBuilder()
-        } else {
-            CosmasProto.FileCemetery.parseFrom(cemeteryBytes.getContent()).toBuilder()
-        }
-        val tombs = cemetery.cemeteryList.toMutableList()
-        tombs.removeIf { it.fileId == request.oldFileId }
         try {
-            this.storage.create(
-                    getBlobInfo(cemeteryName, request.info),
-                    cemetery.clearCemetery().addAllCemetery(tombs).build().toByteArray())
+            mediator.withWriteCemetery(request.info) { cemetery ->
+                val cemeteryBuilder = cemetery.toBuilder()
+                val tombs = cemeteryBuilder.cemeteryList.toMutableList()
```

Comment:
This probably needs some comments as well.

Reply:
done